unit_tests/test_setup.py

```diff
@@ -52,12 +51,17 @@ class InstallResourcesTests(CharmTest):
 
     def setUp(self):
         super().setUp()
-        self.root_dir = self.useFixture(TempDir())
+        self.root_dir = self.fakes.fs.root
+        os.makedirs(self.root_dir.join('etc/cron.d'))
 
         patcher_chown = mock.patch('os.chown')
         patcher_chown.start()
         self.addCleanup(patcher_chown.stop)
 
+        patcher_pwnam = mock.patch('pwd.getpwnam')
+        mock_pwnam = patcher_pwnam.start()
+        mock_pwnam.return_value = mock.MagicMock(pw_uid=456)
```

Comment:
does it really have to be a `MagicMock`? i.e. are you using __ methods? I'd just go with

`mock_pwnam.return_value.pw_uid=456`